grails-app/services/org/pih/warehouse/inventory/InventoryService.groovy

```diff
@@ -1384,9 +1384,26 @@ class InventoryService implements ApplicationContextAware {
             Map<String, AvailableItem> availableItems = productAvailabilityService.getAvailableItemsAtDateAsMap(
                     currentLocation,
                     [cmd.product],
-                    cmd.transactionDate
+                    new Date(cmd.transactionDate.time + 2000)
             )
 
+            // 1. Create the baseline transaction
+            if (isInventoryBaselineEnabled) {
+                // We'd have weird behaviour if we allowed two transactions to exist at the same exact time (precision at the
+                // database level is to the second) so fail if there's already a transaction on the items for the given date.
+                List<InventoryItem> inventoryItems = availableItems.values().collect{ it.inventoryItem }
+                if (hasTransactionEntriesOnDate(currentLocation, adjustmentTransactionDate, inventoryItems)) {
```

Comment:
This is looking closer to what I was expecting but still a question:

Why put this check for hasTransactionEntriesOnDate inside the isInventoryBaselineEnabled block? Shouldn't we do it even if baseline transactions are disabled?

That's why I was suggesting to put it in `recordStockProductInventoryTransactionService.createAdjustmentTransaction`. That way we only check for duplicates at adjustmentTransactionDate if we actually have an adjustment transaction to create.

Reply:
I moved the validation outside the isInventoryBaselineEnabled block. I am not moving it to the `createAdjustmentTransaction` function because the code shouldn't be executed when there is a validation error. So, I am just validating it before proceeding deeper into that function.